src/test/java/org/opensearch/sdk/TestNetty4Transport.java

```diff
@@ -25,11 +25,13 @@
 public class TestNetty4Transport extends OpenSearchTestCase {
 
     private ThreadPool threadPool;
-    private NettyTransport nettyTransport = new NettyTransport();
+    private ExtensionsRunner extensionsRunner;
```

Comment:
Can we use `TestExtensionsRunner` here @dbwiddis?

Reply:
Yes, we can, our problem here is synchronizing all the tests so they aren't all using the transport service associated with it.  I've suggested to @mloufra to use [`@ResourceLock`](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/api/parallel/ResourceLock.html).

Reply:
> Yes, we can, our problem here is synchronizing all the tests so they aren't all using the transport service associated with it. I've suggested to @mloufra to use [`@ResourceLock`](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/api/parallel/ResourceLock.html).

Since the error message is from `TransportCommunicationIT`, `TestNetty4Transport` and `TestExtensionsRunner`, I added `@ResourceLock("transportService")` in each test class, but still cause same error.